### DISCUSSION TITLE
user_defines.txt vs git

### DISCUSSION THREAD

**Discussion by LupusTheCanine** [ExpressLRS/ExpressLRS#3026] [Ideas]
Please move user_defines.txt to something like default_user_defines.txt and make user_defines.txt not tracked. Having user_defines.txt constantly overwritten is PITA.

**Comment by mha1** [comment]
Don't edit user_defines.txt. Create a new file super_defines.txt and add or change existing defines. super_defines.txt is gitignored so won't be uploaded or overwritten by pulling the repo. Example:

super_defines.txt
```
#-DRegulatory_Domain_EU_CE_2400
-DRegulatory_Domain_ISM_2400

#-DRegulatory_Domain_FCC_915
#-DRegulatory_Domain_EU_868

-DMY_BINDING_PHRASE="xxxx"
-DHOME_WIFI_SSID="yyy"
-DHOME_WIFI_PASSWORD="zzz"
-DLOCK_ON_FIRST_CONNECTION

-DUART_INVERTED

!-DAUTO_WIFI_ON_INTERVAL="40"
```

Note: the "!" alters existing defines

**Reply by LupusTheCanine** [reply]
thanks, it would be nice if it was described in the wiki

**Comment by mha1** [comment]
https://github.com/ExpressLRS/Docs/pull/441
